src/pybind/mgr/dashboard/services/auth/auth.py

```diff
@@ -81,20 +82,16 @@ class Local(BaseAuth):
     LOGOUT_URL = '#/login'
     sso = False
 
-    @classmethod
-    def get_auth_name(cls):
-        return cls.__name__.lower()
-
-    def to_dict(self) -> 'BaseAuth.Config':
+    @staticmethod
+    def to_dict() -> 'BaseAuth.Config':
         return BaseAuth.Config()
 
     @classmethod
     def from_dict(cls, s_dict: BaseAuth.Config) -> 'Local':
-        # pylint: disable=unused-argument
-        return cls()
+        raise NotImplementedError
 
 
-class SSOAuth(BaseAuth):
+class SSOAuthMixin(BaseAuth):
```

Comment:
Mixins don't usually derive from ABC, they're helper classes that bring misc methods.
```suggestion
class SSOAuthMixin:
```